File: BleModule/src/main/java/com/mpdc4gsr/ble/EventObserver.java
package com.mpdc4gsr.ble;

import android.bluetooth.BluetoothAdapter;
import android.bluetooth.BluetoothDevice;

import com.mpdc4gsr.commons.observer.Observer;

import java.util.UUID;



public interface EventObserver extends Observer {

    default void onBluetoothAdapterStateChanged(int state) {
    }


    default void onCharacteristicRead(Request request, byte[] value) {
    }


    default void onCharacteristicChanged(Device device, UUID service, UUID characteristic,
                                         byte[] value) {
    }


    default void onCharacteristicWrite(Request request, byte[] value) {
    }


    default void onRssiRead(Request request, int rssi) {
    }


    default void onDescriptorRead(Request request, byte[] value) {
    }


    default void onNotificationChanged(Request request, boolean isEnabled) {
    }


    default void onMtuChanged(Request request, int mtu) {
    }


    default void onPhyChange(Request request, int txPhy, int rxPhy) {
    }


    default void onRequestFailed(Request request, int failType, Object value) {
    }


    default void onConnectionStateChanged(Device device) {
    }


    default void onConnectFailed(Device device, int failType) {
    }


    default void onConnectTimeout(Device device, int type) {
    }
}

File: BleModule/src/main/java/com/mpdc4gsr/ble/EventObserver.kt
package com.mpdc4gsr.ble

import com.mpdc4gsr.commons.observer.Observer
import java.util.UUID

interface EventObserver : Observer {
    fun onBluetoothAdapterStateChanged(state: Int) {
    }

    fun onCharacteristicRead(request: Request?, value: ByteArray?) {
    }

    fun onCharacteristicChanged(
        device: Device?, service: UUID?, characteristic: UUID?,
        value: ByteArray?
    ) {
    }

    fun onCharacteristicWrite(request: Request?, value: ByteArray?) {
    }

    fun onRssiRead(request: Request?, rssi: Int) {
    }

    fun onDescriptorRead(request: Request?, value: ByteArray?) {
    }

    fun onNotificationChanged(request: Request?, isEnabled: Boolean) {
    }

    fun onMtuChanged(request: Request?, mtu: Int) {
    }

    fun onPhyChange(request: Request?, txPhy: Int, rxPhy: Int) {
    }

    fun onRequestFailed(request: Request?, failType: Int, value: Any?) {
    }

    fun onConnectionStateChanged(device: Device?) {
    }

    fun onConnectFailed(device: Device?, failType: Int) {
    }

    fun onConnectTimeout(device: Device?, type: Int) {
    }
}
